diff -up dpdk-1.8.0/mk/rte.lib.mk.libext dpdk-1.8.0/mk/rte.lib.mk
--- dpdk-1.8.0/mk/rte.lib.mk.libext	2015-02-05 11:22:23.994597860 +0200
+++ dpdk-1.8.0/mk/rte.lib.mk	2015-02-05 11:26:38.784937405 +0200
@@ -35,11 +35,16 @@ include $(RTE_SDK)/mk/internal/rte.clean
 include $(RTE_SDK)/mk/internal/rte.build-pre.mk
 include $(RTE_SDK)/mk/internal/rte.depdirs-pre.mk
 
+# All libraries have the same ABI version for now. Upstream library versions
+# start at 1, use 0RH to make sure we dont clash and make it clear this
+# is not upstream version.
+LIBABIVER=0RH
+
 # VPATH contains at least SRCDIR
 VPATH += $(SRCDIR)
 
 ifeq ($(RTE_BUILD_SHARED_LIB),y)
-LIB := $(patsubst %.a,%.so,$(LIB))
+LIB := $(patsubst %.a,%.so.$(LIBABIVER),$(LIB))
 endif
 
 _BUILD = $(LIB)
@@ -76,7 +76,7 @@ O_TO_A_DO = @set -e; \
 	$(O_TO_A) && \
 	echo $(O_TO_A_CMD) > $(call exe2cmd,$(@))
 
-O_TO_S = $(LD) $(_CPU_LDFLAGS) -shared $(OBJS-y) -o $(LIB)
+O_TO_S = $(LD) $(_CPU_LDFLAGS) -shared $(OBJS-y) -Wl,-soname,$(LIB) -o $(LIB)
 O_TO_S_STR = $(subst ','\'',$(O_TO_S)) #'# fix syntax highlight
 O_TO_S_DISP = $(if $(V),"$(O_TO_S_STR)","  LD $(@)")
 O_TO_S_DO = @set -e; \
@@ -159,9 +164,13 @@ endif
 # install lib in $(RTE_OUTPUT)/lib
 #
 $(RTE_OUTPUT)/lib/$(LIB): $(LIB)
+	$(eval LIBSONAME := $(basename $(LIB)))
 	@echo "  INSTALL-LIB $(LIB)"
 	@[ -d $(RTE_OUTPUT)/lib ] || mkdir -p $(RTE_OUTPUT)/lib
 	$(Q)cp -f $(LIB) $(RTE_OUTPUT)/lib
+ifeq ($(RTE_BUILD_SHARED_LIB),y)
+	$(Q)ln -s -f $< $(RTE_OUTPUT)/lib/$(LIBSONAME)
+endif
 
 #
 # Clean all generated files
