diff -up dpdk-2.0.0-1872.git1c29883c/examples/Makefile.jobstats-ex dpdk-2.0.0-1872.git1c29883c/examples/Makefile
--- dpdk-2.0.0-1872.git1c29883c/examples/Makefile.jobstats-ex	2015-02-25 10:49:24.850891254 +0200
+++ dpdk-2.0.0-1872.git1c29883c/examples/Makefile	2015-02-25 10:49:30.124876602 +0200
@@ -52,7 +52,6 @@ DIRS-y += ipv4_multicast
 DIRS-$(CONFIG_RTE_LIBRTE_KNI) += kni
 DIRS-y += l2fwd
 DIRS-$(CONFIG_RTE_LIBRTE_IVSHMEM) += l2fwd-ivshmem
-DIRS-$(CONFIG_RTE_LIBRTE_JOBSTATS) += l2fwd-jobstats
 DIRS-y += l3fwd
 DIRS-$(CONFIG_RTE_LIBRTE_ACL) += l3fwd-acl
 DIRS-$(CONFIG_RTE_LIBRTE_POWER) += l3fwd-power
