diff -up dpdk-2.1.0-2898.gitf1e779ec/examples/ip_pipeline/Makefile.buildopts dpdk-2.1.0-2898.gitf1e779ec/examples/ip_pipeline/Makefile
--- dpdk-2.1.0-2898.gitf1e779ec/examples/ip_pipeline/Makefile.buildopts	2015-08-11 23:36:41.000000000 +0300
+++ dpdk-2.1.0-2898.gitf1e779ec/examples/ip_pipeline/Makefile	2015-08-12 12:46:00.212761965 +0300
@@ -70,7 +70,7 @@ SRCS-$(CONFIG_RTE_LIBRTE_PIPELINE) += pi
 SRCS-$(CONFIG_RTE_LIBRTE_PIPELINE) += pipeline_routing.c
 
 CFLAGS += -I$(SRCDIR) -I$(SRCDIR)/pipeline
-CFLAGS += -O3
+CFLAGS += -O3 -mavx2
 CFLAGS += $(WERROR_FLAGS) -Wno-error=unused-function -Wno-error=unused-variable
 
 include $(RTE_SDK)/mk/rte.extapp.mk
