diff -up dpdk-1.8.0/lib/librte_eal/linuxapp/eal/Makefile.dtneeded dpdk-1.8.0/lib/librte_eal/linuxapp/eal/Makefile
--- dpdk-1.8.0/lib/librte_eal/linuxapp/eal/Makefile.dtneeded	2014-12-20 01:38:39.000000000 +0200
+++ dpdk-1.8.0/lib/librte_eal/linuxapp/eal/Makefile	2015-02-03 13:33:33.510099351 +0200
@@ -49,6 +49,8 @@ CFLAGS += -I$(RTE_SDK)/lib/librte_pmd_af
 CFLAGS += -I$(RTE_SDK)/lib/librte_pmd_xenvirt
 CFLAGS += $(WERROR_FLAGS) -O3
 
+LDLIBS += -ldl -lpthread
+
 # specific to linuxapp exec-env
 SRCS-$(CONFIG_RTE_LIBRTE_EAL_LINUXAPP) := eal.c
 SRCS-$(CONFIG_RTE_LIBRTE_EAL_LINUXAPP) += eal_hugepage_info.c
diff -up dpdk-1.8.0/lib/librte_pmd_pcap/Makefile.dtneeded dpdk-1.8.0/lib/librte_pmd_pcap/Makefile
--- dpdk-1.8.0/lib/librte_pmd_pcap/Makefile.dtneeded	2014-12-20 01:38:39.000000000 +0200
+++ dpdk-1.8.0/lib/librte_pmd_pcap/Makefile	2015-02-03 13:33:51.859056765 +0200
@@ -37,6 +37,8 @@ include $(RTE_SDK)/mk/rte.vars.mk
 #
 LIB = librte_pmd_pcap.a
 
+LDLIBS += -lpcap
+
 CFLAGS += -O3
 CFLAGS += $(WERROR_FLAGS)
 
diff -up dpdk-1.8.0/lib/librte_vhost/Makefile.dtneeded dpdk-1.8.0/lib/librte_vhost/Makefile
--- dpdk-1.8.0/lib/librte_vhost/Makefile.dtneeded	2014-12-20 01:38:39.000000000 +0200
+++ dpdk-1.8.0/lib/librte_vhost/Makefile	2015-02-03 13:33:33.510099351 +0200
@@ -30,6 +30,7 @@
 #   OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 include $(RTE_SDK)/mk/rte.vars.mk
+LDIBS += -lfuse
 
 # library name
 LIB = librte_vhost.a
diff -up dpdk-1.8.0/mk/rte.lib.mk.dtneeded dpdk-1.8.0/mk/rte.lib.mk
--- dpdk-1.8.0/mk/rte.lib.mk.dtneeded	2014-12-20 01:38:39.000000000 +0200
+++ dpdk-1.8.0/mk/rte.lib.mk	2015-02-03 13:33:33.510099351 +0200
@@ -76,7 +76,7 @@ O_TO_A_DO = @set -e; \
 	$(O_TO_A) && \
 	echo $(O_TO_A_CMD) > $(call exe2cmd,$(@))
 
-O_TO_S = $(LD) $(_CPU_LDFLAGS) -shared $(OBJS-y) -o $(LIB)
+O_TO_S = $(LD) $(_CPU_LDFLAGS) -shared $(LDLIBS) $(OBJS-y) -Wl,-soname,$(LIB) -o $(LIB)
 O_TO_S_STR = $(subst ','\'',$(O_TO_S)) #'# fix syntax highlight
 O_TO_S_DISP = $(if $(V),"$(O_TO_S_STR)","  LD $(@)")
 O_TO_S_DO = @set -e; \
