diff -up dpdk-2.0.0/lib/librte_eal/linuxapp/eal/Makefile.dtneeded dpdk-2.0.0/lib/librte_eal/linuxapp/eal/Makefile
--- dpdk-2.0.0/lib/librte_eal/linuxapp/eal/Makefile.dtneeded	2015-04-03 17:13:45.000000000 +0300
+++ dpdk-2.0.0/lib/librte_eal/linuxapp/eal/Makefile	2015-04-17 15:23:38.267984467 +0300
@@ -53,6 +53,8 @@ CFLAGS += -I$(RTE_SDK)/drivers/net/af_packet
 CFLAGS += -I$(RTE_SDK)/lib/librte_ivshmem
 CFLAGS += $(WERROR_FLAGS) -O3
 
+LDLIBS += -ldl -lpthread
+
 # specific to linuxapp exec-env
 SRCS-$(CONFIG_RTE_LIBRTE_EAL_LINUXAPP) := eal.c
 SRCS-$(CONFIG_RTE_LIBRTE_EAL_LINUXAPP) += eal_hugepage_info.c
diff -up dpdk-2.1.0/drivers/net/pcap/Makefile.dtneeded dpdk-2.0.0/lib/librte_pmd_pcap/Makefile
--- dpdk-2.1.0/drivers/net/pcap/Makefile.dtneeded	2015-04-03 17:13:45.000000000 +0300
+++ dpdk-2.1.0/drivers/net/pcap/Makefile	2015-04-17 15:23:38.267984467 +0300
@@ -37,6 +37,8 @@ include $(RTE_SDK)/mk/rte.vars.mk
 #
 LIB = librte_pmd_pcap.a
 
+LDLIBS += -lpcap
+
 CFLAGS += -O3
 CFLAGS += $(WERROR_FLAGS)
 
diff -up dpdk-2.0.0/lib/librte_vhost/Makefile.dtneeded dpdk-2.0.0/lib/librte_vhost/Makefile
--- dpdk-2.0.0/lib/librte_vhost/Makefile.dtneeded	2015-04-03 17:13:45.000000000 +0300
+++ dpdk-2.0.0/lib/librte_vhost/Makefile	2015-04-17 15:24:37.593834737 +0300
@@ -44,6 +44,7 @@ CFLAGS += -I vhost_user
 else
 CFLAGS += -I vhost_cuse -lfuse
 LDFLAGS += -lfuse
+LDLIBS += -lfuse
 endif
 
 ifeq ($(CONFIG_RTE_LIBRTE_VHOST_NUMA),y)
